service/src/main/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsService.java

```diff
@@ -17,22 +16,20 @@
 import uk.gov.hmcts.reform.fpl.model.Orders;
 import uk.gov.hmcts.reform.fpl.model.common.DocumentBundle;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisNoticeOfProceeding;
 import uk.gov.hmcts.reform.fpl.service.time.Time;
 import uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper;
 
 import java.time.format.FormatStyle;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
-import static uk.gov.hmcts.reform.fpl.enums.DocmosisImages.COURT_SEAL;
-import static uk.gov.hmcts.reform.fpl.enums.DocmosisImages.CREST;
 import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
 
 @Service
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
-public class NoticeOfProceedingsService {
+public class NoticeOfProceedingsService extends DocmosisTemplateDataGeneration<DocmosisNoticeOfProceeding> {
```

Comment:
could we have this in a separate template service to separation of duties? So the template generation service only handles the generation of the template for notice of proceedings and `NoticeOfProceedingsService` just has general methods needed for notice of proceedings.

Reply:
agree, done.